include/dkm_utils.hpp

```diff
@@ -124,4 +125,44 @@ std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> get_best_means(
 	return best_means;
 }
 
+
+
+
+    /**
+	 * Return the best clustering obtained from a given number of k-means
+	 * calculations.
+	 *
+	 * @param points		 Sequence of points to be clustered.
+	 * @param parameters	 Clustering related parameters.
+	 * @param n_init		 Number of times a k-means clustering will be calculated.
+	 *
+	 * @return Clustering with the lowest inertia.
+	 */
+
+
+	template <typename T, size_t N>
+	std::tuple<std::vector<std::array<T, N>>, std::vector<uint32_t>> get_best_means_parallel(
+	const std::vector<std::array<T, N>>& points, const clustering_parameters<T>& parameters, uint32_t n_init = 10) {
+
+		// seed generator once 
+		std::random_device rand_device;
+		uint64_t seed = parameters.has_random_seed() ? parameters.get_random_seed() : rand_device();
+		CustomGenerator rand_engine(seed);
```

Comment:
There is another issue with the `CustomGenerator` type here which is not recognized due to the circular dependency between the `dkm_utils.hpp` and `dkm_parallel.hpp` files. Apart from the `CustomGenerator`, there is no reason to include the `dkm_parallel.hpp` file at the `dkm_utils.hpp` and since the `CustomGenerator` should be used in both files, I propose the following solution:
- transfer the `CustomGenerator` declaration at the `dkm_utils.hpp` file
- remove the `#include "dkm_parallel.hpp"` from the `dkm_utils.hpp` file